62.CAD/Polyline.h

```diff
@@ -255,10 +281,221 @@ class CPolyline
 		NOTE that PolylineConnectors are special object types, user does not add them and they should not be part of m_sections vector
 	*/
 
+	void preprocessPolylineWithStyle(const CPULineStyle& lineStyle)
+	{
+		PolylineConnectorBuilder connectorBuilder;
+
+		float phaseShiftTotal = lineStyle.phaseShift;
+		for (uint32_t sectionIdx = 0u; sectionIdx < m_sections.size(); sectionIdx++)
+		{
+			const auto& section = m_sections[sectionIdx];
+
+			if (section.type == ObjectType::LINE)
+			{
+				// calculate phase shift at each point of each line in section
+				const uint32_t linePointCnt = section.count + 1u;
+				for (uint32_t i = 0u; i < linePointCnt; i++)
+				{
+					const uint32_t currIdx = section.index + i;
+					auto& linePoint = m_linePoints[currIdx];
+					if (i == 0u)
+					{
+						linePoint.phaseShift = phaseShiftTotal;
+						continue;
+					}
+
+					const auto& prevLinePoint = m_linePoints[section.index + i - 1u];
+					const float64_t2 line = linePoint.p - prevLinePoint.p;
+					const double lineLen = glm::length(line);
+
+					if (lineStyle.isRoadStyleFlag)
+					{
+						connectorBuilder.addLineNormal(line, lineLen, prevLinePoint.p, phaseShiftTotal);
+					}
+
+					const double changeInPhaseShiftBetweenCurrAndPrevPoint = std::remainder(lineLen, 1.0f / lineStyle.reciprocalStipplePatternLen) * lineStyle.reciprocalStipplePatternLen;
+					linePoint.phaseShift = glm::fract(phaseShiftTotal + changeInPhaseShiftBetweenCurrAndPrevPoint);
+					phaseShiftTotal = linePoint.phaseShift;
+				}
+			}
+			else if (section.type == ObjectType::QUAD_BEZIER)
+			{
+				// calculate phase shift at point P0 of each bezier
+				const uint32_t quadBezierCnt = section.count;
+				const bool isLastSection = (sectionIdx == m_sections.size() - 1u);
+
+				for (uint32_t i = 0u; i <= quadBezierCnt; i++)
```

Comment:
I'm a little confused, shouldn't this be `i < quadBezierCnt` or `i <= quadBezierCnt-1`?
and last bezier in section is `i == quadBezierCnt-1` ?

make sure to test a case where you got a polyline with only line a 2-3 connected beziers in the middle